### PR TITLE
First Real Test of API and Supporting Utilities

### PR DESCRIPTION
**Pull request by thudsonbu**
This PR includes the first testing utility function for a POST request (and supporting functions) as well as the first two tests for our api. These tests check to make sure that when valid credentials sent to the `user/register` route that the response is a success and that when invalid credentials are sent the response is a failure. 

We will want to refactor to seperate files for different controllers eventually but this is a good starting point.

You can try this out by using two terminals. In the first terminal run `NODE_ENV=test npm run dev` to start the api and connect it to the `agend-test` database (as opposed to the `agenda` database). Then, in the second terminal run `npm run test` which will show the results of the tests. Currently the tests hang after being completed and I am not entirely sure why but I put a ticket in for that.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 const assert  = require("assert");
```
Comment (zbarnz) [inline test/test.js:1]: We will need to add mocha to the list of environments in `.eslintrc` or Eslint will get mad at this file

**Comment by zbarnz** [comment]
> You can try this out by using two terminals. In the first terminal run `NODE_ENV=test npm run dev` to start the api and connect it to the `agend-test` database (as opposed to the `agenda` database). Then, in the second terminal run `npm run test` which will show the results of the tests. Currently the tests hang after being completed and I am not entirely sure why but I put a ticket in for that.

For windows peasants  I think we will need to use a script to start api and connect to db. I added `"winTest": "set NODE_ENV=test&&npm run dev"` to my scripts in `package.json`. Maybe there is a better way?

**Comment by zbarnz** [comment]
We will also need the private key to test

**Comment by thudsonbu** [comment]
@zbarnz I think you might not be using bash as your terminal. Bash allows you to run those commands.

**Pull request merged**